lib/inspec/profile.rb

```diff
@@ -19,29 +19,27 @@ def self.from_path(path, options = nil)
     attr_reader :path
     attr_reader :metadata
 
+    # rubocop:disable Metrics/AbcSize
     def initialize(options = nil)
       @options = options || {}
-
-      @params = {}
       @logger = options[:logger] || Logger.new(nil)
-
       @path = @options[:path]
-      fail 'Cannot read an empty path.' if @path.nil? || @path.empty?
-      fail "Cannot find directory #{@path}" unless File.directory?(@path)
-
-      @metadata = read_metadata
-      @params = @metadata.params
-      # use the id from parameter, name or fallback to nil
-      @profile_id = options[:id] || params[:name] || nil
-      @params[:name] = @profile_id
 
-      @params[:rules] = rules = {}
       @runner = Runner.new(
         id: @profile_id,
         backend: :mock,
         test_collector: @options.delete(:test_collector),
       )
-      @runner.add_tests([@path], @options)
+
+      @options[:ignore_supports] = true
+      tests, libs, metadata = @runner.add_tests([@path], @options)
+      @content = tests + libs + metadata
+
+      # NB if you want to check more than one profile, use one
+      # Inspec::Profile#from_file per profile
+      m = metadata.first
+      @metadata = Metadata.from_ref(m[:ref], m[:content], @profile_id, @logger)
```

Comment:
imho parse metadata beore adding/parsing tests; if anything fails with metadata, we don't jump into parsing tests